include/ucl.h

```diff
@@ -825,11 +826,14 @@ typedef void* ucl_object_iter_t;
  * @param iter opaque iterator, must be set to NULL on the first call:
  * ucl_object_iter_t it = NULL;
  * while ((cur = ucl_iterate_object (obj, &it)) != NULL) ...
+ * @param ep pointer record exception (such as ENOMEM), could be NULL
  * @return the next object or NULL
  */
-UCL_EXTERN const ucl_object_t* ucl_object_iterate (const ucl_object_t *obj,
-		ucl_object_iter_t *iter, bool expand_values);
+UCL_EXTERN const ucl_object_t* ucl_object_iterate2 (const ucl_object_t *obj,
```

Comment:
The naming is a bit unfortunate for the public API. Maybe it is better to name this function like `ucl_object_iterate_with_error`? I know it is a bit verbose but it is much more clear than `ucl_object_iterate2` from my point of view.